src/WorldIDIdentityManagerImplV3.sol

```diff
@@ -0,0 +1,233 @@
+///      to perform the insertion using EIP-4844.
+/// @dev This is the implementation delegated to by a proxy.
+contract WorldIDIdentityManagerImplV3 is WorldIDIdentityManagerImplV2 {
+    ///////////////////////////////////////////////////////////////////////////////
+    ///                   A NOTE ON IMPLEMENTATION CONTRACTS                    ///
+    ///////////////////////////////////////////////////////////////////////////////
+
+    // This contract is designed explicitly to operate from behind a proxy contract. As a result,
+    // there are a few important implementation considerations:
+    //
+    // - All updates made after deploying a given version of the implementation should inherit from
+    //   the latest version of the implementation. This contract inherits from its previous implementation
+    //   WorldIDIdentityManagerImplV2. This prevents storage clashes.
+    // - All functions that are less access-restricted than `private` should be marked `virtual` in
+    //   order to enable the fixing of bugs in the existing interface.
+    // - Any function that reads from or modifies state (i.e. is not marked `pure`) must be
+    //   annotated with the `onlyProxy` and `onlyInitialized` modifiers. This ensures that it can
+    //   only be called when it has access to the data in the proxy, otherwise results are likely to
+    //   be nonsensical.
+    // - This contract deals with important data for the WorldID system. Ensure that all newly-added
+    //   functionality is carefully access controlled using `onlyOwner`, or a more granular access
+    //   mechanism.
+    // - Do not assign any contract-level variables at the definition site unless they are
+    //   `constant`.
+    //
+    // Additionally, the following notes apply:
+    //
+    // - Initialisation and ownership management are not protected behind `onlyProxy` intentionally.
+    //   This ensures that the contract can safely be disposed of after it is no longer used.
+    // - Carefully consider what data recovery options are presented as new functionality is added.
+    //   Care must be taken to ensure that a migration plan can exist for cases where upgrades
+    //   cannot recover from an issue or vulnerability.
+
+    ///////////////////////////////////////////////////////////////////////////////
+    ///                    !!!!! DATA: DO NOT REORDER !!!!!                     ///
+    ///////////////////////////////////////////////////////////////////////////////
+
+    // To ensure compatibility between upgrades, it is exceedingly important that no reordering of
+    // these variables takes place. If reordering happens, a storage clash will occur (effectively a
+    // memory safety error).
+
+    /// @notice The table of verifiers for verifying batch identity insertions following the EIP-4844 scheme.
+    VerifierLookupTable internal batchInsertion4844Verifiers;
```

Comment:
We should add methods to set a batchInsertion4844Verifiers after initialization.

For consistency I would also add a get method like [here](https://github.com/reilabs/world-id-contracts/blob/a894eb66408b0a035f30dfdab2498387c08386e5/src/WorldIDIdentityManagerImplV2.sol#L165)

Reply:
They are used to set a new verifier if you need to change (e.g. re-run trusted setup etc)